lib/iris/tests/integration/test_pickle.py

```diff
@@ -31,35 +31,43 @@
     from iris.fileformats.grib.message import GribMessage
 
 
+class Common(object):
+    def pickle_cube(self, path):
+        # Ensure that data proxies are pickleable.
+        cube = iris.load(path)[0]
+        with self.temp_filename('.pkl') as filename:
+            with open(filename, 'wb') as f:
+                pickle.dump(cube, f)
+            with open(filename, 'r') as f:
+                ncube = pickle.load(f)
+        self.assertEqual(ncube, cube)
+
+
 @tests.skip_data
 @tests.skip_grib
-class TestGribMessage(tests.IrisTest):
-    def test(self):
-        # Check that a GribMessage pickles without errors.
-        path = tests.get_data_path(('GRIB', 'fp_units', 'hours.grib2'))
-        messages = GribMessage.messages_from_filename(path)
-        message = next(messages)
+class TestGribMessage(Common, tests.IrisTest):
+    def setUp(self):
+        self.path = tests.get_data_path(('GRIB', 'fp_units', 'hours.grib2'))
+
+    def pickle_obj(self, obj):
         with self.temp_filename('.pkl') as filename:
             with open(filename, 'wb') as f:
-                pickle.dump(message, f)
+                pickle.dump(obj, f)
+
+    def test(self):
+        # Check that a GribMessage pickles without errors.
+        messages = GribMessage.messages_from_filename(self.path)
+        obj = next(messages)
+        self.pickle_obj(obj)
```

Comment:
@cpelley Is the test simply that there are no exceptions raised by `pickle.dump()` within `pickle_obj()` ? 

There are no explicit `assert`s associated with this test nor `test_data` ...